core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -812,10 +812,13 @@ private[spark] object MapOutputTracker extends Logging {
         logError(errorMessage)
         throw new MetadataFetchFailedException(shuffleId, startPartition, errorMessage)
       } else {
+        var totalSize = 0L
         for (part <- startPartition until endPartition) {
-          splitsByAddress.getOrElseUpdate(status.location, ArrayBuffer()) +=
-            ((ShuffleBlockId(shuffleId, mapId, part), status.getSizeForBlock(part)))
+          totalSize += status.getSizeForBlock(part)
```

Comment:
This can be simplified like: `val totalSize = (startPartition until endPartition).map(status.getSizeForXXX).sum`.